src/pybind/rados/rados.pyx

```diff
@@ -3564,6 +3573,69 @@ returned %d, but should return zero on success." % (self.name, ret))
             raise make_ex(ret, "Failed to notify %r" % (obj))
         return True
 
+    @requires(('obj', str), ('oncomplete', opt(Callable)), ('msg', str), ('timeout_ms', int))
+    def aio_notify(self, obj, oncomplete, msg='', timeout_ms=5000):
+        """
+        Asynchronously send a rados notification to an object
+
+        :param obj: the name of the object to notify
+        :type obj: str
+        :param oncomplete: what to do when notification acknowledgement is received
+                           or notification times out
+        :type oncomplete: completion
+        :param msg: optional message to send in the notification
+        :type msg: str
+        :param timeout_ms: notify timeout (in ms)
+        :type timeout_ms: int
+
+        :raises: :class:`Error`
+        :returns: completion object
+        """
+        self.require_ioctx_open()
+
+        msglen = len(msg)
+        obj = cstr(obj, 'obj')
+        msg = cstr(msg, 'msg')
+
+        cdef:
+            Completion completion
+            char *_obj = obj
+            char *_msg = msg
+            int _msglen = msglen
+            uint64_t _timeout_ms = timeout_ms
+            char *reply
+            size_t replylen = 0
+
+        def oncomplete_(completion_v):
+            cdef:
+                Completion _completion_v = completion_v
+                notify_ack_t *acks = NULL
+                notify_timeout_t *timeouts = NULL
+                size_t nr_acks
+                size_t nr_timeouts
+            return_value = _completion_v.get_return_value()
+            if return_value == 0:
+                return_value = rados_decode_notify_response(reply, replylen, &acks, &nr_acks, &timeouts, &nr_timeouts)
+                rados_buffer_free(reply)
+            if return_value == 0:
+                ack_list = [(ack.notifier_id, ack.cookie, '' if not ack.payload_len \
+                                                             else ack.payload[:ack.payload_len]) for ack in acks[:nr_acks]]
+                timeout_list = [(timeout.notifier_id, timeout.cookie) for timeout in timeouts[:nr_timeouts]]
+                rados_free_notify_response(acks, nr_acks, timeouts)
+                return oncomplete(_completion_v, 0, ack_list, timeout_list)
```

Comment:
Nit: should the format (and expected argument format) of the callback function be specified?

Reply:
Probably. However, I'm not sure how to go about verifying that? Or, are you suggesting to document it?

Reply:
Just suggesting you document it.

Reply:
ack -- will push the changes.